packages/server/migrations/201908131532000-table-global-settings.js

```diff
@@ -0,0 +1,54 @@
+module.exports = {
+  up: async function (queryInterface, Sequelize) {
+    try {
+      await queryInterface.sequelize.query(`BEGIN`);
+
+      await queryInterface.createTable('GlobalSettings', {
+        id: {
+          allowNull: false,
+          primaryKey: true,
+          type: Sequelize.UUID
+        },
+        name: {
+          type: Sequelize.STRING,
+          allowNull: false,
+          unique: true,
+        },
```

Comment:
What is the name? I'm not sure this is needed.

Reply:
I do think that we need to create this table for a common purpose. Storing one row and one file in the table is bad practice and something strange. One step of extending functionality of the table is to add "name" column to make it possible to identify rows .

Reply:
Could you give me an example of what a few rows would be? 

Having a one-row table for common app settings, that are going to change (unlike Env vars), seems pretty reasonable to me.

Reply:
Oh, do you mean storing ALL settings in one row? In one field?

My view is to store settings like this:
"Data.json" --> JSON of the Data.json
"notificationEnabled" --> true
"smtpServer" ---> host (string)
and others, and only after this structure we are able to add for instance "isPublic" column or similar.

So if we add only one row, then to prevent access issues we will need to add some flags right into JSON. For instance, "{dataJson: {}, dataJsonPublic: true, anotherSettings: 1, anotherSetting2: 2} "

Reply:
And this is only first case where we can fall down. There are 5 rules of DB normalization, and one of them is to put different things in different rows.

Reply:
I mean storing them all in one row, but in multiple fields. So in that example, we'd have a column for each item. 

Can we prevent access issues on a graphQL side? As in, it's easy to get the info from the db, but it won't actually send it to a client?

I'm imagining, similarly, that in the graphQL side, we don't need to know these are rows. We could just have a query for something like,

```graphQL
globalSettings{
  entityGraph
  notificationsEnabled
  smtpServer
}
```

Reply:
So there are two approaches. You propose to store settings in one row with multiple columns and I propose to store named setting in multiple rows with. Got your point, thanks. Yes, now I  understood that in my approach to get all settings of the application we would send a lot of queries. 


Thanks, going to change it a bit.

Reply:
Sounds good. I'm not really sure, but for one, we could always change later.

Reply:
Never mind, my misunderstanding.